src/tokenOperations/tokenOperationsFactory.ts

```diff
@@ -0,0 +1,550 @@
+        return this.setSpecialRoleOnSemiFungible(args);
+    }
+
+    setSpecialRoleOnNonFungible(args: INonFungibleSetSpecialRoleArgs): Transaction {
+        const parts = [
+            "setSpecialRole",
+            utf8ToHex(args.tokenIdentifier),
+            addressToHex(args.user),
+            ...(args.addRoleNFTCreate ? [utf8ToHex("ESDTRoleNFTCreate")] : []),
+            ...(args.addRoleNFTBurn ? [utf8ToHex("ESDTRoleNFTBurn")] : []),
+            ...(args.addRoleNFTUpdateAttributes ? [utf8ToHex("ESDTRoleNFTUpdateAttributes")] : []),
+            ...(args.addRoleNFTAddURI ? [utf8ToHex("ESDTRoleNFTAddURI")] : []),
+            ...(args.addRoleESDTTransferRole ? [utf8ToHex("ESDTTransferRole")] : []),
+        ];
+
+        return this.createTransaction({
+            sender: args.manager,
+            receiver: this.config.esdtContractAddress,
+            nonce: args.transactionNonce,
+            gasPrice: args.gasPrice,
+            gasLimitHint: args.gasLimit,
+            executionGasLimit: this.config.gasLimitSetSpecialRole,
+            dataParts: parts
+        });
+    }
+
+    nftCreate(args: INFTCreateArgs): Transaction {
+        const parts = [
+            "ESDTNFTCreate",
+            utf8ToHex(args.tokenIdentifier),
+            bigIntToHex(args.initialQuantity),
+            utf8ToHex(args.name),
+            bigIntToHex(args.royalties),
+            utf8ToHex(args.hash),
+            bufferToHex(args.attributes),
+            ...args.uris.map(utf8ToHex),
+        ];
+
+        // Note that the following is an approximation (a reasonable one):
+        const nftData = args.name + args.hash + args.attributes + args.uris.join("");
+        const storageGasLimit = nftData.length * this.config.gasLimitStorePerByte.valueOf();
+
+        return this.createTransaction({
+            sender: args.creator,
+            receiver: args.creator,
+            nonce: args.transactionNonce,
+            gasPrice: args.gasPrice,
+            gasLimitHint: args.gasLimit,
+            executionGasLimit: this.config.gasLimitESDTNFTCreate.valueOf() + storageGasLimit.valueOf(),
+            dataParts: parts
+        });
+    }
+
+    pause(args: IPausingArgs): Transaction {
+        if (args.pause == args.unpause) {
+            throw new Err("Must set either 'pause' or 'unpause' (one and only one should be set).")
+        }
+
+        const parts = [
+            args.pause ? "pause" : "unPause",
+            utf8ToHex(args.tokenIdentifier)
+        ];
+
+        return this.createTransaction({
+            sender: args.manager,
+            receiver: this.config.esdtContractAddress,
+            nonce: args.transactionNonce,
+            gasPrice: args.gasPrice,
+            gasLimitHint: args.gasLimit,
+            executionGasLimit: this.config.gasLimitPausing,
+            dataParts: parts
+        });
+    }
+
+    freeze(args: IFreezingArgs): Transaction {
```

Comment:
the `wipe` functionality is missing

Reply:
Added :+1: 

(a few others are missing, as well - e.g. "add NFT URI")